CMakeLists.txt

```diff
@@ -371,6 +371,22 @@ if(SLANG_ENABLE_SLANGC)
     )
 endif()
 
+#
+# WebAssembly bindings for Slang
+#
+# This is an executable target because emcmake produces .a files without bindings if you just create a static library
+# https://stackoverflow.com/questions/63622009/static-library-built-with-cmake-as-a-with-emscripten-instead-of-wasm-js
+slang_add_target(
+  source/slang-wasm
+  EXECUTABLE
+  EXCLUDE_FROM_ALL
+  USE_FEWER_WARNINGS
+  LINK_WITH_PRIVATE miniz lz4_static slang core compiler-core
+  INCLUDE_DIRECTORIES_PUBLIC include source/slang-wasm
+)
```

Comment:
Do we want to `INSTALL` or `INSTALL_COMPONENT` this target?

Reply:
> Do we want to `INSTALL` or `INSTALL_COMPONENT` this target?

I probably don't understand the CMake concept of "instsall" but I don't think we want this.
At the moment the way I work with this component is to just build slang-wasm.js and slang-wasm.wasm and then copying it to the web server root directory for use in the demo app.
This works fine IMO.

If we wanted to put this into some more official release package I think we have some JS API design work to do first.

Reply:
I want this to be in the release package even when the API isn't full designed. This is necessary for setting up CI to building our web demos. The CI needs  to download slang-wasm from a release package and automatically deploy it to the web site.

Reply:
> I want this to be in the release package even when the API isn't full designed. This is necessary for setting up CI to building our web demos. The CI needs to download slang-wasm from a release package and automatically deploy it to the web site.

Right, at some point releasing to the website needs to not involve any extra steps...
In that case we may want to also consider adding some slang-wasm tests at some point -- should be quite easy with node-JS, which comes with emsdk.
It seems very possible that a working native Slang would for some reason not work in Wasm.